File: Code Samples/OOP/OOP II - Harsh Duche/Abstraction.java

abstract class GdscMember {
    public abstract void id();
}

class DevTeam extends GdscMember {
    @Override
    public void id() {
        System.out.println("My name is Harsh & I'm development team member");
    }
}

public class Abstraction {
    public static void main(String[] args) {
        DevTeam dev1 = new DevTeam();
        dev1.id();
    }
}

File: Code Samples/OOP/OOP II - Harsh Duche/Python/Abstraction.py
# Abstract class = a class which contains one or more abstract methods.
# Abstract method = a method that has a declaration but does not have an implementation.

# prevents a user from creating an object of that class
# + compels a user to override abstract methods in a child class

from abc import ABC, abstractclassmethod

class gdsc_member(ABC):
    @abstractclassmethod
    def id(self):
        pass


class dev_team(gdsc_member):
    def id(self):
        print("My name is Harsh & I'm development team member")

dev1 = dev_team()

dev1.id()